test/CowAmmModule/CreateAmm.sol

```diff
@@ -0,0 +1,172 @@
+        emit CowAmmModule.CowAmmCreated(safe, token0, token1, preCalculatedOrderHash);
+
+        bytes32 orderHash = cowAmmModule.createAmm(
+            ammData.token0,
+            ammData.token1,
+            ammData.minTradedToken0,
+            address(ammData.priceOracle),
+            ammData.priceOracleData,
+            ammData.appData
+        );
+
+        assertEq(address(eHandler.domainVerifiers(safe, domainSeparator)), address(composableCow));
+        assertEq(token0.allowance(address(safe), address(relayer)), type(uint256).max);
+        assertEq(token1.allowance(address(safe), address(relayer)), type(uint256).max);
+        assertTrue(composableCow.singleOrders(address(safe), orderHash));
+        assertEq(preCalculatedOrderHash, orderHash);
+        assertEq(cowAmmModule.activeOrders(safe), orderHash);
+    }
+
+    function testRevertIfNoToken0Balance() public {
+        _testRevertIfNoTokenBalance(address(token0));
+    }
+
+    function testRevertIfNoToken1Balance() public {
+        _testRevertIfNoTokenBalance(address(token1));
+    }
+
+    function testRevertIfAmmAlreadyExists() public {
+        setUpDefaultCowAmm();
+
+        ConstantProduct.Data memory ammData = getDefaultData();
+
+        vm.prank(address(safe));
+        vm.expectRevert(abi.encodeWithSelector(CowAmmModule.ActiveAMM.selector));
+        cowAmmModule.createAmm(
+            ammData.token0,
+            ammData.token1,
+            ammData.minTradedToken0,
+            address(ammData.priceOracle),
+            ammData.priceOracleData,
+            ammData.appData
+        );
+    }
+
+    function testFallbackHandlerSetDoesNotSetAgain() public {
+        setUpDefaultSafe();
+
+        ConstantProduct.Data memory ammData = getDefaultData();
+
+        vm.mockCallRevert(
+            address(safe),
+            abi.encodeWithSelector(FallbackManager.setFallbackHandler.selector),
+            abi.encode("called setFallbackHandler")
+        );
+        vm.prank(address(safe));
+        vm.expectRevert();
+        cowAmmModule.createAmm(
+            ammData.token0,
+            ammData.token1,
+            ammData.minTradedToken0,
+            address(ammData.priceOracle),
+            ammData.priceOracleData,
+            ammData.appData
+        );
+    }
+
+    function testDomainVerifierSetDoesNotSetAgain() public {
+        setUpDefaultSafe();
+
+        ConstantProduct.Data memory ammData = getDefaultData();
+
+        vm.mockCallRevert(
+            address(eHandler),
+            abi.encodeWithSelector(SignatureVerifierMuxer.setDomainVerifier.selector),
+            abi.encode("called setDomainVerifier")
+        );
+        vm.prank(address(safe));
+        vm.expectRevert();
+        cowAmmModule.createAmm(
+            ammData.token0,
+            ammData.token1,
+            ammData.minTradedToken0,
+            address(ammData.priceOracle),
+            ammData.priceOracleData,
+            ammData.appData
+        );
+    }
```

Comment:
Same issue as in the test before: we should set the fallback handler ourselves, then set the domain verifier, then do everything as in this test and _not_ expect that it reverts.